db/hints/manager.cc

```diff
@@ -190,6 +190,10 @@ future<> manager::wait_for_sync_point(abort_source& as, const sync_point::shard_
         }
     });
 
+    if (as.abort_requested()) {
+        local_as.request_abort();
```

Comment:
Why can't we use one abort source instead of propagating? Happy to commit this fix first, but we should avoid manually chaining abort sources.

Reply:
I think we can. However, I thought that propagating aborts downwards but not upwards (which would happen if we used only one abort_source) is more composable - an error in function F will abort its sub-routines, but the caller of F might want to handle the error gracefully instead of aborting everything.

Reply:
I don't understand. The top-level caller passes an abort_source. The intermediate-level passes the abort_source to the low-level functions. How is it different from chaining? except for having less code.

Reply:
If the operation is only aborted at the top level, then I agree. However, the abort source can be triggered at the mid level - e.g. one of the parallel invocations spawned in mid-level failed and other invocations should be aborted before returning from mid-level with an error. If we didn't use chaining, the abort at the mid level would be visible at the top level. This might not be what the top-level caller wants, e.g. they would like to ignore or retry in case of some errors.

Here, top-level is [`storage_proxy::wait_for_hint_sync_point`](https://github.com/piodul/scylla/blob/91163fcfa5842aaf725a68a10bb95a4baabe6492/service/storage_proxy.cc#L5348) and mid-level is `manager::wait_for_sync_point`.

To be fair, in this particular case it wouldn't matter if we use chaining or not - if one of the parallel invocations in `storage_proxy::wait_for_sync_point` fails with an error, the top-level abort source is always triggered.